frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -1260,6 +1260,24 @@ export class ExpressionDimension extends Dimension {
       semantic_type = base_type;
     }
 
+    // FIX
+    // when the app uses dimension().field().dimension().field()
+    // the last generated field is badly formatted
```

Comment:
minor nitpick, but I feel like this comment doesn't really resolve any confusion around the code being added here. in fact, it might be a cause for more confusion (future developer may ask, "why are we calling `dimension().field().dimension().field()` and why is it wrong?).

I think something like, "if a dimension has access to a question with result metadata, try to find the field there so that we don't have to try to infer field metadata from the expression," offers a bit more clarity around why these lines of code exist (and from this a future developer might ask, "why would we want to avoid inferring?" and the answer is a bit more obvious: because we might fail at inferring).